### PR TITLE
Add Github workflow 

### LINKED ISSUE #31: Create github workflow
Create GitHub workflow for below scenarios
1. Run syntax checks for .py files for every pull-request.
2. Publish package to pip whenever there is a new release published in GitHub.

### PR DESCRIPTION
**Pull request by rubyruins**
# Description:

Adds Github actions for:
1. Running syntax checks for .py files withevery pull request. 
2. Publishing the package to PyPi whenever a new release is published in Github (via pushing tagged commit.)

Fixes #31 

## Type of change:

- [x] New feature (non-breaking change which adds functionality)
 
# Checklist:

- [x] I have performed a self-review of my own code.
- [x] My changes generate no new warnings.

## Screenshots:

image
image
image

### CODE REVIEW

File: .github/workflows/publish_pypi.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+on: push
```
Comment (chebroluharika) [inline .github/workflows/publish_pypi.yml:1]: Please change this event to "release on created mode"

File: .github/workflows/publish_pypi.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to PyPI
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
```
Comment (chebroluharika) [inline .github/workflows/publish_pypi.yml:12]: @yvkrishna Can you please confirm which python version you used to develop this code

File: .github/workflows/publish_pypi.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+    - name: Publish distribution 📦 to PyPI
+      if: startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (chebroluharika) [inline .github/workflows/publish_pypi.yml:31]: @yvkrishna please add PYPI_API_TOKEN in secrets section

**Comment by chebroluharika** [comment]
@rubyruins Please attach screenshot for pypi publish action.
To test that workflow, you need to fork this repo, create your own test pypi account, give those credentials in secrets section in your forked repo. Then create a release in GitHub, check if workflow gets triggered and publishes package to Pypi

File: .github/workflows/publish_pypi.yml (hunk 1 above)
Comment (rubyruins) [reply .github/workflows/publish_pypi.yml:1]: Oh, okay. I actually thought it works when a tagged commit (for release) is pushed, but I will change it to the new one now 👍

**Comment by rubyruins** [comment]
@chebroluharika, sure, give me some time, I'll test it out.

**Comment by rubyruins** [comment]
@chebroluharika, I've updated the PR and made the necessary changes. Could you please check?

**Comment by chebroluharika** [comment]
LGTM
@yvkrishna Please add PYPI_USERNAME and PYPI_PASSWORD in secrets section. Once, you add them, I will merge this PR.

**Comment by yvkrishna** [comment]
Yeah Done

**Pull request merged**